keps/sig-storage/1432-volume-health-monitor/README.md

```diff
@@ -674,8 +679,200 @@ _This section must be completed when targeting alpha to a release._
+    to check the volume health. It is configurable with 1 minute as default. Will consider
+    changing it to 5 minutes by default to avoid overloading the K8s API server.
+    When scaled out across many nodes, low frequency checks can still produce high volumes of
+    events. To control this, we should use options on the eventrecorder to control QPS per key.
+    This way we can collapse keys and have a slow update cadence per key.
+  - API calls that may be triggered by changes of some Kubernetes resources
+    (e.g. update of object X triggers new updates of object Y)
+  - periodic API calls to reconcile state (e.g. periodic fetching state,
+    heartbeats, leader election, etc.)
+
+* **Will enabling / using this feature result in introducing new API types?**
+  Describe them, providing:
+  - API type: No
+  - Supported number of objects per cluster: No
+  - Supported number of objects per namespace (for namespace-scoped objects): No
+
+* **Will enabling / using this feature result in any new calls to the cloud
+provider?**
+  No.
+
+* **Will enabling / using this feature result in increasing size or count of
+the existing API objects?**
+  Describe them, providing:
+  - API type(s): No
+  - Estimated increase in size: (e.g., new annotation of size 32B):
+    No
+  - Estimated amount of new objects: (e.g., new Object X for every existing Pod)
+    The controller reports events on PVC while the agent reports events on Pod. They work independently of each other. It is recommended that CSI driver should not report duplicate information through the controller and the agent. For example, if the controller detects a failure on one volume, it should record just one event on one PVC. If an agent detects a failure, it should record an event on every pod used by the affected PVC.
+
+    Recovery event will be reported once if the volume condition changes from abnormal back to normal.
+
+* **Will enabling / using this feature result in increasing time taken by any
+operations covered by [existing SLIs/SLOs]?**
+  Think about adding additional work or introducing new steps in between
+  (e.g. need to do X to start a container), etc. Please describe the details.
+  This feature will periodically query storage systems to get the latest volume conditions. So this will have an impact on the performance of the operations running on the storage systems.
+
+* **Will enabling / using this feature result in non-negligible increase of
+resource usage (CPU, RAM, disk, IO, ...) in any components?**
+  Things to keep in mind include: additional in-memory state, additional
+  non-trivial computations, excessive access to disks (including increased log
+  volume), significant amount of data sent and/or received over network, etc.
+  This through this both in small and large cases, again with respect to the
+  [supported limits].
+  This will increase load on the storage systems as it periodically queries them.
```

Comment:
Given this, it would be good to include an escape valve to turn this off at runtime. Getting stuck in I/O sleep can take out an entire node.